src/com/facebook/buck/apple/AppleBundle.java

```diff
@@ -866,4 +942,46 @@ public BuildRule getBinaryBuildRule() {
     }
     return ImmutableSortedSet.of();
   }
+
+
+  @Override
+  public Iterable<? extends NativeLinkable> getNativeLinkableDeps(CxxPlatform cxxPlatform) {
+    return ImmutableSet.of();
+  }
+
+  @Override
+  public Iterable<? extends NativeLinkable> getNativeLinkableExportedDeps(CxxPlatform cxxPlatform) {
+    return ImmutableSet.of();
+  }
+
+  @Override
+  public NativeLinkableInput getNativeLinkableInput(
+      CxxPlatform cxxPlatform, Linker.LinkableDepType type) throws NoSuchBuildTargetException {
+    ImmutableList.Builder<Arg> linkerArgsBuilder = ImmutableList.builder();
+    linkerArgsBuilder.addAll(StringArg.from(
```

Comment:
The problem is that when specifying it here, you do not know what kind of bundle you end up in. Right not this is configured like an exported linker flag: the dependency framework tells the binary that when it links with it that the parent should also set an rpath value so it can find the framework with dyld. 

Problem is that the dependency itself doesn't know if the binary that calls the `getNativeLinkableInput` is a binary that will go into an osx app bundle, framework bundle or ios app bundle. Based on that info these paths should be different. That's why I don't think this setting can work correctly from here. I'll try to see if I can modify the binary's linker flags from `AppleBundleDescription`.